### ISSUE TITLE
ansible/rackhd_local.yml fails on downloading static image files

### ISSUE THREAD

**Issue by StevenLee1** [RackHD/RackHD#253]
According to the "Ubuntu Source Based Installation", when I was running the step: "ansible-playbook -i "local," -K -c local rackhd_local.yml", the terminal outputs the wrong message,
"failed: [local] => (item=initrd.img-3.16.0-25-generic) => {"dest": "/root/src/on-http/static/http/common/initrd.img-3.16.0-25-generic", "failed": true, "item": "initrd.img-3.16.0-25-generic", "response": "Request failed: <urlopen error timed out>", "state": "absent", "status_code": -1, "url": "https://bintray.com/artifact/download/rackhd/binary/builds/initrd.img-3.16.0-25-generic"}
msg: Request failed
",
I use " ping https://bintray.com/artifact/download/rackhd/binary/builds/initrd.img-3.16.0-25-generic ",and I got " unknown host", so what should I do?

**Comment by StevenLee1** [comment]
“ping www.google.com” and other websites is ok, but except this website:https://bintray.com/artifact/download/rackhd/binary/builds/initrd.img-3.16.0-25-generic

**Comment by heckj** [comment]
The playbooks are downloading static binary files that are build from code in the repository on-imagebuilder. You can view the files for that within Bintray's site at https://bintray.com/rackhd/binary/on-imagebuilder/1.0/view#files, or more specifically at https://bintray.com/rackhd/binary/on-imagebuilder/1.0/view#files/builds

The file in question is listed, but I'm not able to explicitly attempt to download it while I'm writing this message. I'll check in first thing tomorrow when I'm at a computer - worst case we can regenerate it and upload it, but I would fully expect it (and all the others) to be available from bintray.

**Comment by heckj** [comment]
It looks like Bintray may have changes how their downloads work - that definitely used to work, but now in the UI I'm seeing very different URI structures.

the location `https://dl.bintray.com/rackhd/binary/builds/:initrd.img-3.16.0-25-generic` is getting the right file, so I think the ansible likely needs to be updated to leverage a new base location in the role. I'll set up a PR with what I've learned

**Comment by benbp** [comment]
@StevenLee1 I changed the issue title to be a little bit more descriptive. Looks like @heckj's update at https://github.com/RackHD/RackHD/pull/255 should fix and will be merged soon.

**Comment by heckj** [comment]
The PR I set up doesn't appear to be correct, and in fact breaks things - the original download URLs are working through ansible in my testing. I'm going to close that PR, as it's not doing anything useful there.

@StevenLee1 try a `ping bintray.com` and make sure you can get to bintray, that may be part of what's breaking. I don't know what I did yesterday that convinced me things were broken, but reviewing it this morning it's looking like it's all working correctly.

For my testing, I ran `wget https://bintray.com/artifact/download/rackhd/binary/builds/initrd.img-3.16.0-25-generic`, as well as running the current ansible project from within a fresh Ubuntu 14.04 VM through to success.

**Comment by heckj** [comment]
Ah, I figured out what I did wrong - the link into bintray (which Bintray wants us to use) ends in a redirect mechanism to where they host files on a CDN. Monday morning, I tried the command with `curl` with the `-L/--location` flag to ask it to follow redirects (or the variants, `--post301`, `--post302` and `--post303`). `wget` does this implicitly. For example:

`curl -L https://bintray.com/artifact/download/rackhd/binary/builds/initrd.img-3.16.0-25-generic -o initrd.img-3.16.0-25-generic` works as expected.
